File: 05-01-databinding/java-05-01-databinding/app/src/main/java/com/example/databinding/MainActivity.java
package com.example.databinding;

import androidx.databinding.DataBindingUtil;
import androidx.databinding.ObservableField;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import com.example.databinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ViewModel viewModel;
//    private int count = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        viewModel = new ViewModel();
        binding.setViewModel(viewModel);
        viewModel.title.set("Hello Android");
    }

//    public void sayHello(View view) {
//        count++;
//        viewModel.title.set("안녕하세요. " + count + "번째 클릭입니다.");
//    }

    public static class ViewModel {
        private int count = 0;

        public ObservableField<String> title = new ObservableField<>();

        public void sayHello() {
            count++;
            title.set("안녕하세요. " + count + "번째 클릭입니다.");
        }
    }
}

File: 05-01-databinding/kotlin-05-01-databinding/app/src/main/java/com/example/databinding/MainActivity.kt
package com.example.databinding

import androidx.databinding.DataBindingUtil
import androidx.databinding.ObservableField
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.Toast

import com.example.databinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private val viewModel: ViewModel by lazy {
        ViewModel()
    }
//    private var count = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.viewModel = viewModel
        viewModel.title.set("Hello Android")
    }

//    fun sayHello(view: View) {
//        count++
//        viewModel.title.set("안녕하세요. " + count + "번째 클릭입니다.")
//    }

    inner class ViewModel {
        private var count = 0

        var title = ObservableField<String>()

        fun sayHello() {
            count++
            title.set("안녕하세요. " + count + "번째 클릭입니다.")
        }
    }
}
